rct229/rules/section22/section22rule8.py

```diff
@@ -0,0 +1,145 @@
+MIN_CHW_PRIMARY_LOOP_COOLING_CAPACITY = 300.0 * ureg("ton")
+
+
+class Section22Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule8, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule8.PrimaryFluidLoopRule(),
+            index_rmr="baseline",
+            id="22-8",
+            description="For Baseline chilled water system with cooling capacity of 300 tons or more, the secondary pump shall be modeled with variable-speed drives.",
+            list_path="ruleset_model_instances[0].fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list containing all HVAC systems that are modeled in the rmi_b
+        available_type_list = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_list
+                ]
+            )
+            and len(primary_secondary_loop_dict) > 0
+        )
+
+    def create_data(self, context, data):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+        loop_pump_dict = {
+            pump["loop_or_piping"]: pump for pump in find_all("$.pumps[*]", rmi_b)
+        }
+
+        chw_loop_capacity_dict = {}
+        for chiller in find_all("$.chillers[*]", rmi_b):
+            cooling_loop_id = chiller["cooling_loop"]
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[cooling_loop_id] = ZERO.POWER
+            chw_loop_capacity_dict[cooling_loop_id] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+        primary_loop_ids = primary_secondary_loop_dict.keys()
+
+        child_loop_speed_control_dict = {}
+        for primary_loop_id in primary_loop_ids:
+            for child_loop in find_all(
+                f'$.fluid_loops[*][?(@.id="{primary_loop_id}")].child_loops[*]', rmi_b
+            ):
+                child_loop_speed_control_dict[child_loop["id"]] = getattr_(
+                    loop_pump_dict[child_loop["id"]],
+                    "speed_control",
+                    "speed_control",
+                )
+
+        return {
+            "loop_pump_dict": loop_pump_dict,
```

Comment:
After walk through the logic, I dont really think the `child_loop_speed_control_dict` is necessary. Does the RDS require this variable?

Either `loop_pump_dictionary` or `child_loop_speed_control_dict` should work in the rule checking. and it seems `loop_pump_dictionary` is not used beyond line 95.